src/GeocodeJSON_Reader_Rapid_DOM.cpp

```diff
@@ -0,0 +1,317 @@
+        }
+        
+        // GeocodeJSON soecifies geometry as Lon-Lat
+        m_geometry . SetLongitude ( coords [ 0 ] . GetDouble () );
+        m_geometry . SetLatitude ( coords [ 1 ] . GetDouble ());
+        
+        return & m_geometry; 
+    } 
+
+    virtual const char* Id () const     { return GetStringProperty ( "id", true ); }
+    virtual const char* Layer () const  { return GetStringProperty ( "layer", true ); }
+    virtual const char* Source () const { return GetStringProperty ( "source" ); }
+     
+    virtual uint32_t AccuracyMeters () const 
+    { 
+        const Value& g = GetMember ( "properties" );
+        if ( ! g . IsObject () )
+        {
+            throw std :: logic_error ( "Property \"properties\" is not an object" ); 
+        }
+        bool found;
+        const Value& v = GetOptionalMember ( g, "accuracy", found );
+        if ( ! found )
+        {
+            return 0; 
+        }
+        if ( ! v . IsUint () )
+        {
+            throw std :: logic_error ( string ( "Property \"" ) + "accuracy" +"\" is not an unsigned integer" ); 
+        }
+        return v . GetUint (); 
+    }
+    
+    virtual double Confidence () const 
+    { 
+        const Value& g = GetMember ( "properties" );
+        if ( ! g . IsObject () )
+        {
+            throw std :: logic_error ( "Property \"properties\" is not an object" ); 
+        }
+        bool found;
+        const Value& v = GetOptionalMember ( g, "confidence", found );
+        if ( ! found )
+        {
+            return 0; 
+        }
+        if ( ! v . IsDouble () )
+        {
+            throw std :: logic_error ( string ( "Property \"" ) + "confidence" +"\" is not a number" ); 
+        }
+        return v . GetDouble (); 
+    }
+    
+    virtual const char* Label () const  { return GetStringProperty ( "label" ); }
+    virtual const char* Name () const   { return GetStringProperty ( "name" ); }
+    
+    virtual const char* HouseNumber () const    { return GetStringProperty ( "housenumber" ); }
+    virtual const char* Street () const         { return GetStringProperty ( "street" ); }
+    virtual const char* Postcode () const       { return GetStringProperty ( "postcode" ); }
+    virtual const char* City () const           { return GetStringProperty ( "city" ); }
+    virtual const char* District () const       { return GetStringProperty ( "district" ); }
+    virtual const char* County () const         { return GetStringProperty ( "county" ); }
+    virtual const char* Region () const         { return GetStringProperty ( "region" ); }
+    virtual const char* RegionAbbr () const     { return GetStringProperty ( "region_abbr", true ); }
+    virtual const char* Country () const        { return GetStringProperty ( "country" ); }
+    virtual const char* CountryAbbr () const    { return GetStringProperty ( "country_abbr" ); }
+    
+    virtual const char* Admin ( unsigned int p_level ) const 
```

Comment:
consider making this logic table-driven; wait for resolution of GeocodeJSON issue.